### PR TITLE
fix(schematics): ng-add should pass if sourceRoot is relative to workspace root

### LINKED ISSUE #3052: ng g @angular/fire:ng-add fails in a project where sourceRoot starts with the project's root
### Version info

**Angular:** 12.2.0

**Firebase:** latest

**AngularFire:** latest

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**

### How to reproduce these conditions

This can be easily reproduced in Nx by following these steps, but would occur in a regular Angular project if the angular.json file had a similar project configuration to below. Reproduction steps for Nx are found in the Issue mention, for angular just `ng new` and update the configuration.

### Debug output

** Screenshots **
image

### Expected behavior

`ng add @angular/fire` passes if sourceRoot and root are both relative to the workspace root

### Actual behavior

`ng add @angular/fire` only passes if sourceRoot is relative to root

----

In a workspace with a project configured as below, `ng add @angular/fire` fails due to the sourceRoot path being appended to the root, rather than replacing it.

```json
{
  "root": "apps/admin"
  "sourceRoot": "apps/admin/src"
}
```

https://github.com/angular/angularfire/blob/5ecf8757d5ea5cbd93e0c0881a95deea9f0310e3/src/schematics/setup/index.ts#L36

Maybe this portion of code could use something like
```typescript
const sourcePath = project.sourceRoot?.startsWith(project.root) 
		? project.sourceRoot 
		: [project.root, project.sourceRoot].filter(it => !!it).join('/'); 
```

I'd be happy to put in a PR to make the change if its agreed upon.

### PR DESCRIPTION
**Pull request by AgentEnder**
### Checklist

   - Issue number for this PR: #3052
   - Docs included?: no
   - Test units included?: yes
   - In a clean directory, `yarn install`, `yarn test` run successfully? yes

### Description

Closes #3052

### CODE REVIEW

File: src/schematics/setup/index.ts [hunk 1]
```diff
@@ -33,7 +33,9 @@ export const setupProject =
 
     const { project, projectName } = getProject(config, tree);
 
-    const sourcePath = [project.root, project.sourceRoot].filter(it => !!it).join('/');
+    const sourcePath = project.sourceRoot?.startsWith(project.root)
+      ? project.sourceRoot
+      : [project.root, project.sourceRoot].filter(it => !!it).join('/');
```
Comment (jamesdaniels) [inline src/schematics/setup/index.ts:38]: Now I'm doubting myself, wonder if this logic is wrong entirely... perhaps all this should be `project.sourceRoot || project.root` instead if they're both relative. WDYT?

Comment (AgentEnder) [reply src/schematics/setup/index.ts:38]: In Nx land I can confirm that we always treat both of these as workspace root, but I'm not sure if thats the case for all regular angular workspaces. I was trying to keep this from being a "breaking" fix at all, I think just `project.sourceRoot ?? project.root` would be much simpler, if you are good with that fix.

Comment (jamesdaniels) [reply src/schematics/setup/index.ts:38]: Lets go for that. I think in coding this I misunderstood the paths, I was pretty rushed to get the schematics out the door.

Comment (AgentEnder) [reply src/schematics/setup/index.ts]: Sure, I just pushed up those changes.

**Review by jamesdaniels — APPROVED** [review]
LGTM, thanks

**Pull request merged**